backend/src/main/java/com/bakdata/conquery/io/result/csv/CsvRenderer.java

```diff
@@ -48,12 +61,52 @@ public void printLine(PrintSettings cfg, EntityPrintId entity, List<ResultInfo>
 		writer.addValues((Object[]) entity.getExternalId());
 		try {
 			for (int i = 0; i < infos.size(); i++) {
-				writer.addValue(infos.get(i).getType().printNullable(cfg, value[i]));
+				final ResultInfo info = infos.get(i);
+
+				final Function3<ResultType, Object, Function<Object, Object>, String>
+						printer =
+						FIELD_MAP.getOrDefault(info.getType().getClass(), this::printDefault);
+				final Function<Object, Object> valueMapper = info.getValueMapper().orElse(Function.identity());
```

Comment:
Sehe das jetzt erst, aber wäre es möglich die Logik von ResultInfo zu printer vorzuziehen und nur als Parameter reinzureichen? Damit könnte man einiges an Lookup etc sparen. Du machst hier ja gerade pro Zeile pro Zelle einen Map Lookup ein Optional auflösen und den invoke auf printer

Reply:
Ich habe die printer Logik jetzt vorgezogen in die `createCSVBody()` so dass in den eigentlichen printern jetzt auch kein Lookup mehr passiert